xarray/convert.py

```diff
@@ -117,7 +119,7 @@ def to_iris(dataarray):
     if 'cell_methods' in dataarray.attrs:
         args['cell_methods'] = parse_cell_methods(dataarray.attrs['cell_methods'])
 
-    cube = iris.cube.Cube(dataarray.to_masked_array(), **args)
+    cube = iris.cube.Cube(ma.masked_invalid(dataarray), **args)
```

Comment:
This converts everything into a dask array, even numpy arrays. Instead, we probably only want to convert xarray objects that already store their data in dask arrays into masked dask arrays. You can dispatch based on the type of `dataarray.data`.

Reply:
would it be worth supporting dask masked arrays in `DataArray.to_masked_array()`?

Reply:
Yes that's probably a good idea.

Reply:
xref: #1769